node/src/utils/specimen.rs

```diff
@@ -0,0 +1,951 @@
+            inactive_validators: vec_prop_specimen(estimator, "validator_count", cache),
+        }
+    }
+}
+
+impl LargestSpecimen for BlockHash {
+    fn largest_specimen<E: SizeEstimator>(estimator: &E, cache: &mut Cache) -> Self {
+        BlockHash::new(LargestSpecimen::largest_specimen(estimator, cache))
+    }
+}
+
+// impls for `casper_hashing`, which is technically a foreign crate -- so we put them here.
+impl LargestSpecimen for Digest {
+    fn largest_specimen<E: SizeEstimator>(_estimator: &E, _cache: &mut Cache) -> Self {
+        // Hashes are fixed size by definition, so any value will do.
+        Digest::hash("")
+    }
+}
+
+impl LargestSpecimen for BlockPayload {
+    fn largest_specimen<E: SizeEstimator>(estimator: &E, cache: &mut Cache) -> Self {
+        // We cannot just use the standard largest speciment for `DeployHashWithApprovals`, as this
+        // would cause a quadratic increase in deploys. Instead, we generate one large deploy that
+        // contains the number of approvals if they are spread out across the block.
+
+        let large_deploy = Deploy::largest_specimen(estimator, cache).with_approvals(
+            btree_set_distinct_from_prop(estimator, "average_approvals_per_deploy_in_block", cache),
+        );
+        let large_deploy_hash_with_approvals = DeployHashWithApprovals::from(&large_deploy);
+
+        let deploys = vec![
+            large_deploy_hash_with_approvals.clone();
+            estimator.require_parameter::<usize>("max_deploys_per_block")
+        ];
+        let transfers = vec![
+            large_deploy_hash_with_approvals;
+            estimator.require_parameter::<usize>("max_transfers_per_block")
+        ];
+
+        BlockPayload::new(
+            deploys,
+            transfers,
+            vec_prop_specimen(estimator, "max_accusations_per_block", cache),
+            LargestSpecimen::largest_specimen(estimator, cache),
+        )
+    }
+}
+
+impl LargestSpecimen for DeployHashWithApprovals {
+    fn largest_specimen<E: SizeEstimator>(estimator: &E, cache: &mut Cache) -> Self {
+        // Note: This is an upper bound, the actual value is lower. We are keeping the order of
+        //       magnitude intact though.
+        let max_items = estimator.require_parameter::<usize>("max_deploys_per_block")
+            + estimator.require_parameter::<usize>("max_transfers_per_block");
+        DeployHashWithApprovals::new(
+            LargestSpecimen::largest_specimen(estimator, cache),
+            btree_set_distinct(estimator, max_items, cache),
+        )
+    }
+}
+
+impl LargestSpecimen for Deploy {
+    fn largest_specimen<E: SizeEstimator>(estimator: &E, cache: &mut Cache) -> Self {
+        Deploy::new(
+            LargestSpecimen::largest_specimen(estimator, cache),
+            LargestSpecimen::largest_specimen(estimator, cache),
+            LargestSpecimen::largest_specimen(estimator, cache),
+            vec![/* Legacy field, always empty */],
```

Comment:
Unfortunately although this _is_ a deprecated field, Deploys are still deemed valid if up to `[deploys.max_dependencies]` hashes are specified here.

Reply:
Let's just fill it with `max_dependencies` deploy hashes then!

Reply:
Fixed in 3393622d0